### PR TITLE
Add transform feedback functions

### PR DESCRIPTION
**Pull request by IcanDivideBy0**
I finally followed what have been done with textures concerning the multiple transform feedback creation. The only function missing is `isTransformFeedback` which is not that usefull IMO.

### CODE REVIEW

**Comment by IcanDivideBy0** [comment]
fixes #101

File: src/native.rs [hunk 1]
```diff
@@ -2123,6 +2124,74 @@ impl HasContext for Context {
...
+        let gl = &self.raw;
+        gl.ResumeTransformFeedback();
+    }
+
+    unsafe fn transform_feedback_varyings(&self, program: Self::Program, varyings: &[&str], buffer_mode: u32) {
+        let gl = &self.raw;
+        gl.TransformFeedbackVaryings(
+            program,
+            varyings.len() as i32,
+            varyings.as_ptr() as *const *const native_gl::types::GLchar,
```
Comment (IcanDivideBy0) [inline src/native.rs:2170]: Please double check this!

File: src/native.rs [hunk 2]
```diff
@@ -2123,6 +2124,74 @@ impl HasContext for Context {
...
+
+        gl.GetTransformFeedbackVarying(
+            program,
+            index,
+            buf_size,
+            std::ptr::null_mut(),
+            size as *mut i32,
+            ty as *mut u32,
+            name.as_mut_ptr() as *mut i8
+        );
```
Comment (IcanDivideBy0) [inline src/native.rs:2191]: And also this!

File: src/lib.rs [hunk 3]
```diff
@@ -900,7 +902,25 @@ pub trait HasContext {
...
+
+    unsafe fn end_transform_feedback(&self);
+
+    unsafe fn pause_transform_feedback(&self);
+
+    unsafe fn resume_transform_feedback(&self);
+
+    unsafe fn transform_feedback_varyings(&self, program: Self::Program, varyings: &[&str], buffer_mode: u32);
+
+    unsafe fn get_transform_feedback_varying(&self, program: Self::Program, index: u32) -> Option<(i32, u32, String)>;
```
Comment (IcanDivideBy0) [inline src/lib.rs:923]: Not sure that returning a tuple is the good choice there... If you have any suggestion?

File: src/lib.rs [hunk 4]
```diff
@@ -73,6 +74,7 @@ pub trait HasContext {
     type Framebuffer: Copy + Clone + Debug + Eq + Hash + Ord + PartialEq + PartialOrd;
     type Renderbuffer: Copy + Clone + Debug + Eq + Hash + Ord + PartialEq + PartialOrd;
     type Query: Copy + Clone + Debug + Eq + Hash + Ord + PartialEq + PartialOrd;
+    type TransformFeedback: Clone + Debug + Eq + Hash + Ord + PartialEq + PartialOrd;
```
Comment (grovesNL) [inline src/lib.rs:77]: We can also add `Copy` here if we're using `WebTransformFeedbackKey`

File: src/native.rs (hunk 1 above)
Comment (grovesNL) [reply src/native.rs:2170]: To have properly null-terminated strings here, we will need to convert each `&str` into a `CString`, then use [`.as_ptr()` on that instead](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.as_ptr) (note the warning about lifetimes). There is a response in https://users.rust-lang.org/t/converting-str-to-const-c-char/23115 which explains this a bit.

File: src/native.rs (hunk 2 above)
Comment (grovesNL) [reply src/native.rs:2191]: We'll need to do the opposite here, the `CString` documentation talks about this a bit, but the idea is that we could copy something like `CString::from_raw` into our own `String`

File: src/lib.rs (hunk 3 above)
Comment (grovesNL) [reply src/lib.rs:923]: We could probably follow the same pattern that we use for [`ActiveUniform` and `ActiveAttribute`](https://github.com/grovesNL/glow/blob/59c017ccb513b7fb6f64acab7da3790a6483acc4/src/lib.rs#L44-L54) – maybe we could add a `ActiveTransformFeedback` struct or similar?

**Comment by IcanDivideBy0** [comment]
- Added `Clone` to `HasContext::TransformFeedback` type.
- Now using CStrings in native impl.

> We could probably follow the same pattern that we use for ActiveUniform and ActiveAttribute – maybe we could add a ActiveTransformFeedback struct or similar?

I'm not sure about this, to me TransformFeedback are more like VertexArrays than Uniforms/Attributes, it's a king of shorthand to call multiple `bind_base_buffer` in a single call. Maybe I'm wrong on this but I don't feel it's necessary to track the active transform feedback since the active vertex arrays are not tracked either.

**Pull request merged**